### PR TITLE
Improve resiliency to formatting JSON in server

### PR DESCRIPTION
**Pull request by pugnascotia**
Split out from #48450. Make a number of changes so that JSON in the server directory is more resilient to automatic formatting. This covers:

   * Reformatting multiline JSON to embed whitespace in the strings
   * Add helper method reformatJson(), to strip whitespace from a JSON
     document using XContent methods. This is sometimes necessary where
     a test is comparing some machine-generated JSON with an expected
     value.

### CODE REVIEW

**Comment by pugnascotia** [comment]
@elasticmachine run elasticsearch-ci/2

File: server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java [hunk 1]
```diff
@@ -161,6 +162,17 @@ public static String convertToJson(BytesReference bytes, boolean reformatJson, X
...
     }
 
+    /**
+     * Accepts a JSON string and reformats it. This is useful where a piece of JSON is formatted for legibility, but needs
+     * to be stripped of unnecessary whitespace e.g. for comparison in a test.
+     * @param json the JSON to format
+     * @return reformatted JSON
+     * @throws IOException if the reformatting fails, e.g. because the JSON is not well-formed
+     */
+    public static String reformatJson(String json) throws IOException {
```
Comment (atorok) [inline server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java]: I think a name that better reflects what the `reformat` actually does would be better, something like `stripWhitespace` maybe ?

Comment (pugnascotia) [reply server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java]: Thanks @atorok, I renamed the method as you suggested.

**Comment by pugnascotia** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by pugnascotia** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Review by rjernst — APPROVED** [review]
LGTM

**Review by alpar-t — APPROVED** [review]

**Pull request merged**